### ISSUE TITLE
More refactoring for project workflow selection

### ISSUE THREAD

**Issue by srallen** [zooniverse/Panoptes-Front-End#3063]
## Expected behavior

~~Try moving the workflow fetch to be a part of the promises in `fetchProjectData`~~ Done in #3545 
## Current behavior

Currently the workflow is fetched in `componentWillReceiveProps` after the project and associated data is fetched and the promise is resolved.
## Additional information
- **Operating system:**
- **Browser:**

**Comment by srallen** [comment]
Workflow needs a listener in case the workflow is made inactive during the classification session so a new active workflow is selected to be worked on.

**Comment by srallen** [comment]
- Once Zooniverse/Panoptes#2054 is done, then the random workflow selection can be selected using a random index of the project links workflow array. `isWorkflowInactive` would change to comparing the selected workflow index to the project links workflow array. 
- Move API calls for selected workflow into a classify page container component and for all active workflows into a home page container component. Scope actual calls to where they are used. This will fix the issue of Talk pages calling for workflows when not necessary.
- Project default workflow id should also check if active. If not active, should it be cleared and/or in the lab, should we prevent project builders from setting a workflow to default if it is not active? (Related to NfN recent bug with `getSelectedWorkflow` and `isWorkflowInactive` looping and crashing the browser).

**Comment by srallen** [comment]
Workflow selection should be abstracted out into a class to handle which workflow is selected. Conditionals are too complicated as is.

**Comment by srallen** [comment]
Closing in favor of open issue in panoptes-javascript-client repo.
